cmake/kokkoskernels_tpls.cmake

```diff
@@ -0,0 +1,82 @@
+
+FUNCTION(TARGET_LINK_FLAGS_PORTABLE LIBRARY)
+  IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13")
+    #great, this works the "right" way
+    TARGET_LINK_OPTIONS(${LIBRARY} ${ARGN})
+  ELSE()
+    #bummer, this works the "hacky" way
+    TARGET_LINK_LIBRARIES(${LIBRARY} ${ARGN})
+  ENDIF()
+ENDFUNCTION(TARGET_LINK_FLAGS_PORTABLE)
+
+FUNCTION(KOKKOSKERNELS_LINK_TPLS LIBRARY)
+  IF (KOKKOSKERNELS_ENABLE_TPL_BLAS)
+    TARGET_LINK_LIBRARIES(${LIBRARY} PUBLIC ${BLAS_LIBRARIES})
+    TARGET_LINK_FLAGS_PORTABLE(${LIBRARY} PUBLIC ${BLAS_LINK_FLAGS})
```

Comment:
I had to add the lines below to get tests to build with BLAS TPL enabled, due to missing symbols for `?gesv` LAPACK routines causing link errors:

```
    TARGET_LINK_LIBRARIES(${LIBRARY} PUBLIC ${LAPACK_LIBRARIES})
    TARGET_LINK_FLAGS_PORTABLE(${LIBRARY} PUBLIC ${LAPACK_LINK_FLAGS})
```